packfile TestCorrectPackWithOneEmptyObject fails with Go 1.27

### Bug Description

This test asserts the exact bytes in a zlib header in a packfile.

Upstream Go CL https://go.dev/cl/707355 improves the performance of compress/flate (used by compress/zlib), which results in different compressed bytes.

The exact byte output of compress/zlib is not guaranteed to remain stable and should be considered an implementation detail. Only the decompressed content is stable.

### go-git Version

053c039455d22205a332126c2b7c69a1da850deb (main)

### Steps to Reproduce

There is no Go 1.27 RC yet, but you can use `golang.org/dl/gotip` to test with tip Go:

```
$ go install golang.org/dl/gotip@latest
$ gotip download
$ gotip test github.com/go-git/go-git/v6/plumbing/format/packfile
--- FAIL: TestEncoderSuite (0.01s)
    --- FAIL: TestEncoderSuite/TestCorrectPackWithOneEmptyObject (0.00s)
        encoder_test.go:77: 
                Error Trace:    /tmp/go-git/plumbing/format/packfile/encoder_test.go:77
                Error:          Not equal: 
                                expected: []byte{0x50, 0x41, 0x43, 0x4b, 0x0, 0x0, 0x0, 0x2, 0x0, 0x0, 0x0, 0x1, 0x10, 0x78, 0x9c, 0x1, 0x0, 0x0, 0xff, 0xff, 0x0, 0x0, 0x0, 0x1, 0x62, 0xa2, 0x7b, 0x2c, 0xa, 0x27, 0x5a, 0xbb, 0x97, 0xdf, 0x66, 0x3d, 0x8, 0x73, 0x2d, 0x9c, 0xfe, 0x30, 0xaf, 0xc8}
                                actual  : []byte{0x50, 0x41, 0x43, 0x4b, 0x0, 0x0, 0x0, 0x2, 0x0, 0x0, 0x0, 0x1, 0x10, 0x78, 0x9c, 0x3, 0x0, 0x0, 0x0, 0x0, 0x1, 0x62, 0xa2, 0x7b, 0x2c, 0xa, 0x27, 0x5a, 0xbb, 0x97, 0xdf, 0x66, 0x3d, 0x8, 0x73, 0x2d, 0x9c, 0xfe, 0x30, 0xaf, 0xc8}
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1,5 +1,5 @@
                                -([]uint8) (len=44) {
                                - 00000000  50 41 43 4b 00 00 00 02  00 00 00 01 10 78 9c 01  |PACK.........x..|
                                - 00000010  00 00 ff ff 00 00 00 01  62 a2 7b 2c 0a 27 5a bb  |........b.{,.'Z.|
                                - 00000020  97 df 66 3d 08 73 2d 9c  fe 30 af c8              |..f=.s-..0..|
                                +([]uint8) (len=41) {
                                + 00000000  50 41 43 4b 00 00 00 02  00 00 00 01 10 78 9c 03  |PACK.........x..|
                                + 00000010  00 00 00 00 01 62 a2 7b  2c 0a 27 5a bb 97 df 66  |.....b.{,.'Z...f|
                                + 00000020  3d 08 73 2d 9c fe 30 af  c8                       |=.s-..0..|
                                 }
                Test:           TestEncoderSuite/TestCorrectPackWithOneEmptyObject
FAIL
FAIL    github.com/go-git/go-git/v6/plumbing/format/packfile    6.031s
FAIL
```

### Additional Information

As far as I can tell, there is no correctness issue here, just a fragile test.

--- a/plumbing/format/packfile/encoder_test.go
+++ b/plumbing/format/packfile/encoder_test.go
@@ -57,24 +57,30 @@ func (s *EncoderSuite) TestCorrectPackWithOneEmptyObject() {
 	h, err := s.enc.Encode([]plumbing.Hash{o.Hash()}, 10)
 	s.NoError(err)
 
-	// + HASH
 	hb := h.Bytes()
 
 	// PACK + VERSION(2) + OBJECT NUMBER(1)
-	expectedResult := make([]byte, 0, 24+len(hb))
-	expectedResult = append(expectedResult, 'P', 'A', 'C', 'K', 0, 0, 0, 2, 0, 0, 0, 1)
+	expectedPrefix := make([]byte, 0, 13)
+	expectedPrefix = append(expectedPrefix, 'P', 'A', 'C', 'K', 0, 0, 0, 2, 0, 0, 0, 1)
 	// OBJECT HEADER(TYPE + SIZE)= 0001 0000
-	expectedResult = append(expectedResult, []byte{16}...)
-
-	// Zlib header
-	expectedResult = append(expectedResult,
-		[]byte{120, 156, 1, 0, 0, 255, 255, 0, 0, 0, 1}...)
-
-	expectedResult = append(expectedResult, hb...)
+	expectedPrefix = append(expectedPrefix, 16)
 
 	result := s.buf.Bytes()
 
-	s.Equal(expectedResult, result)
+	s.True(bytes.HasPrefix(result, expectedPrefix))
+	s.Require().GreaterOrEqual(len(result), len(hb))
+	s.Equal(hb, result[len(result)-len(hb):])
+
+	p, cleanup := packfileFromReader(s, s.buf)
+	defer cleanup()
+
+	decHash, err := p.ID()
+	s.NoError(err)
+	s.Equal(h, decHash)
+
+	decoded, err := p.Get(o.Hash())
+	s.NoError(err)
+	objectsEqual(s, decoded, o)
 }
 
 func (s *EncoderSuite) TestMaxObjectSize() {

